pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -88,6 +88,7 @@ type ClusterBuilder struct {
 	metadataCerts      *metadataCerts        // Client certificates specified in metadata.
 	clusterID          string                // Cluster in which proxy is running.
 	proxyID            string                // Identifier that uniquely identifies a proxy.
+	proxyMetadata      *model.NodeMetadata   // Metadata of the proxy.
```

Comment:
Anything used here needs to be taken into account in buildClusterKey. We won't want the entire proxy config though, just reduce it to a bool

Reply:
IIUC, `buildClusterKey` creates a `clusterCache` which is used to differentiate unique clusters. If the introduced `proxyMetadata` is not incorporated into the `buildClusterKey`, there could potentially be issues where clusters with different `proxyMetadata` values incorrectly share the same key.

I agree that we don't want to bloat the cache by including the entire `proxyMetadata`. Instead, it makes sense to include only the field(s) that directly impact cluster uniqueness or behavior.

Could you help me understand the reasoning behind reducing `proxyMetadata` to a boolean?  What is the bool based on? 

I also don't see `cb.proxyID` included in the `clusterCache` so I am wondering what in the `proxyMetadata` can impact cluster uniqueness or behavior that isn't already captured by the current `clusterCache`

Reply:
> only the field(s) that directly impact cluster uniqueness or behavior.
> proxyMetadata to a boolean

I meant to do what you said basically -- in the key, store "UseStatefulFormatter: bool" basically. I think that should work?

I think I was slightly misleading since I didn't look close enough -- not everything in ClusterBuilder is in the cache key and in ClusterBuilder its fine to have it as-is (note this design is a bit flawed, since it relies on devs carefully ensuring things they access are included in the cache key... but it is what it is)

Reply:
So tldr: 'add UseStatefulFormatter into the cache key (and hash function)'